lib/lhm/retry_helper.rb

```diff
@@ -0,0 +1,43 @@
+require 'retriable'
+
+module Lhm
+  # RetryHelper standardizes the interface for retry behavior in components like
+  # Entangler and AtomicSwitcher.
+  #
+  # It expects the caller to implement:
+  # DEFAULT_MAX_RETRIES
+  # DEFAULT_RETRY_WAIT
+  #
+  # These defaults can be overridden by setting an instance variable with corresponding names:
+  # @max_retries
+  # @retry_wait
+  #
+  # To retry some behavior:
+  # Retriable.retriable(retry_config) do
+  #   @connection.execute(tagged(stmt))
+  # end
+  module RetryHelper
+    def self.included(base)
+      raise(ArgumentError, "#{base} must define DEFAULT_MAX_RETRIES before calling 'include RetryHelper'") unless base.constants.include?(:DEFAULT_MAX_RETRIES)
+      raise(ArgumentError, "#{base} must define DEFAULT_RETRY_WAIT before calling 'include RetryHelper'") unless base.constants.include?(:DEFAULT_RETRY_WAIT)
+    end
+
+    def retry_config
+      {
+        on: {
+          StandardError => [/Lock wait timeout exceeded/]
+        },
+        tries: @max_retries || self.class::DEFAULT_MAX_RETRIES, # number of attempts
+        base_interval: @retry_wait || self.class::DEFAULT_RETRY_WAIT, # initial interval in seconds between tries
+        multiplier: 1.5, # each successive interval grows by this factor
+        rand_factor: 0.25, # percentage to randomize the next retry interval time
+        max_elapsed_time: 900, # max total time in seconds that code is allowed to keep being retried
```

Comment:
I could totally be wrong, but my reading of https://github.com/kamui/retriable/blob/master/lib/retriable.rb#L41 is that this includes _the first try_ of the code -- that is: `start_time` gets set as soon as `#retriable` is called.

One way to verify this in code would be a test using [Timecop](https://github.com/travisjeffery/timecop), but I'm not sure that's necessary here; a second opinion might be enough.

If `max_elapsed_time` _does_ include the initial try, we probably need to set it to `Float::INFINITY` as it looks like it's a mandatory parameter.